modules/cram/templates/whatshap.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+set -euo pipefail
+
+create_ped () {
+  echo -e "!{pedigreeContent}" > "!{pedigree}"
+}
+
+filter_bams () {
+  # filter all reads without readgroups
+  count=$(${CMD_BCFTOOLS} query -l "!{vcf}" | wc -l)
+  if [ "${count}" -gt 1 ]; then
+    for cram in !{crams}; do
+        ${CMD_SAMTOOLS} view -h ${cram} | awk 'substr($0, 1, 1) == "@" || $0 ~ /RG:Z:/' | ${CMD_SAMTOOLS} view -b > "${cram}_filtered.bam"
+        ${CMD_SAMTOOLS} index "${cram}_filtered.bam"
+    done
+  fi
+}
+
+phase_variants () {
+      local args=()
+      local cramAdded=false
+      args+=("--reference" "!{paramReference}" )
+      args+=("--output" "!{vcfOut}")
+      args+=("!{vcf}")
+      # use filtered cram files if multiple samples are present
+      sampleCount=$(${CMD_BCFTOOLS} query -l "!{vcf}" | wc -l)
+      if [ "${sampleCount}" -gt 1 ]; then
+        for cram in !{crams}; do
+            count=$(${CMD_SAMTOOLS} view -c "${cram}_filtered.bam")
+            if [ "${count}" -gt 0 ]; then
+              #do not run Whatshap if no crams remain
+              cramAdded="true"
+              args+=("${cram}_filtered.bam")
+            fi
+        done
+        args+=("--ped" "!{pedigree}") 
+      # assume all reads belong to the sample if only one sample is present
+      else
+        cramAdded="true"
+        args+=("!{crams}")
+        args+=("--ignore-read-groups")
+      fi
+      args+=("--algorithm" "!{algorithm}")
+      args+=("--internal-downsampling" !{internalDownsampling})
+      args+=("--mapping-quality" !{mappingQuality})
+      args+=("--error-rate" !{errorRate})
+      args+=("--maximum-error-rate" !{maximumErrorRate})
+      args+=("--threshold" !{threshold})
+      args+=("--negative-threshold" !{negativeThreshold})
+      args+=("--default-gq" !{defaultGq})
+      args+=("--recombrate" !{recombrate})
+      #args+=("--supplementary-distance" !{supplementaryDistance}) #disabled because of https://github.com/whatshap/whatshap/issues/579
```

Comment:
fixed in 2.4, see https://github.com/whatshap/whatshap/blob/main/CHANGES.rst#v24-2025-01-22
how about upgrading? also fixed crashes and adds support for multi-allelic variants